src/models/generated/ST1A.jl

```diff
@@ -0,0 +1,233 @@
+- `internal::InfrastructureSystemsInternal`: power system internal reference, do not modify
+"""
+mutable struct ST1A <: AVR
+    "Code input for Underexcitization limiter (UEL) entry. Not supported."
+    UEL_inputs::Int64
+    "Code input for Power System Stabilizer (PSS) or (VOS) entry."
+    PSS_inputs::Int64
+    "Regulator input filter time constant in s"
+    Tr::Float64
+    "Voltage error limits (regulator input) (Vi_min, Vi_max)"
+    Vi_lim::Tuple{Float64, Float64}
+    "First regulator denominator (lead) time constant in s"
+    Tc::Float64
+    "First regulator denominator (lag) time constant in s"
+    Tb::Float64
+    "Second regulator denominator (lead) time constant in s"
+    Tc1::Float64
+    "Second regulator denominator (lead) time constant in s"
+    Tb1::Float64
+    "Voltage regulator gain"
+    Ka::Float64
+    "Voltage regulator time constant in s"
+    Ta::Float64
+    "Limits for regulator output (Va_min, Va_max)"
+    Va_lim::Tuple{Float64, Float64}
+    "Limits for exciter output (Vr_min, Vr_max)"
+    Vr_lim::Tuple{Float64, Float64}
+    "Rectifier loading factor proportional to commutating reactance"
+    Kc::Float64
+    "Rate feedback gain"
+    Kf::Float64
+    "Rate feedback time constant in s"
+    Tf::Float64
+    "Exciter output current limiter gain"
+    K_lr::Float64
+    "Exciter output current limit reference"
+    I_lr::Float64
+    "Reference Voltage Set-point"
+    V_ref::Float64
+    ext::Dict{String, Any}
+    "The states are:
+	Vm: Sensed terminal voltage,
+	Vr1: First Lead-lag state,
+	Vr2: Second lead-lag state,
+	Va: Regulator output state,
+	Vr3: Feedback output state"
+    states::Vector{Symbol}
+    n_states::Int64
+    "power system internal reference, do not modify"
+    internal::InfrastructureSystemsInternal
+end
+
+function ST1A(UEL_inputs, PSS_inputs, Tr, Vi_lim, Tc, Tb, Tc1, Tb1, Ka, Ta, Va_lim, Vr_lim, Kc, Kf, Tf, K_lr, I_lr, V_ref=1.0, ext=Dict{String, Any}(), )
+    ST1A(UEL_inputs, PSS_inputs, Tr, Vi_lim, Tc, Tb, Tc1, Tb1, Ka, Ta, Va_lim, Vr_lim, Kc, Kf, Tf, K_lr, I_lr, V_ref, ext, [:Vm, :Vr1, :Vr2, :Va, :Vr3], 5, InfrastructureSystemsInternal(), )
+end
+
+function ST1A(; UEL_inputs, PSS_inputs, Tr, Vi_lim, Tc, Tb, Tc1, Tb1, Ka, Ta, Va_lim, Vr_lim, Kc, Kf, Tf, K_lr, I_lr, V_ref=1.0, ext=Dict{String, Any}(), )
+    ST1A(UEL_inputs, PSS_inputs, Tr, Vi_lim, Tc, Tb, Tc1, Tb1, Ka, Ta, Va_lim, Vr_lim, Kc, Kf, Tf, K_lr, I_lr, V_ref, ext, )
+end
+
+# Constructor for demo purposes; non-functional.
+function ST1A(::Nothing)
+    ST1A(;
+        UEL_inputs=1,
+        PSS_inputs=1,
+        Tr=0,
+        Vi_lim=(0.0, 0.0),
+        Tc=0,
+        Tb=0,
+        Tc1=0,
+        Tb1=0,
+        Ka=0,
+        Ta=0,
+        Va_lim=(0.0, 0.0),
+        Vr_lim=(0.0, 0.0),
+        Kc=0,
+        Kf=0,
+        Tf=0,
+        K_lr=0,
+        I_lr=0,
+        V_ref=0,
+        ext=Dict{String, Any}(),
+    )
+end
+
+"""Get ST1A UEL_inputs."""
+get_UEL_inputs(value::ST1A) = value.UEL_inputs
```

Comment:
do we need these here or are they inputs from an external component?

Reply:
UEL inputs and PSS inputs are the flags used to determine on which the UEL and PSS signals enter in the AVR. This are flags of the AVR model, and hence must be parameters (or flags) of the AVR.

I'm thinking of renaming to UEL_flags and PSS_flags.

Reply:
See ST1A in here:
https://www.powerworld.com/files/Block-Diagrams-17.pdf

Reply:
ok agreed this needs to be a parameter.